examples/3D_contour.py

```diff
@@ -0,0 +1,153 @@
+def _logistics4(x, a=1, b=1, c=-1, d=1):
+    return a + b / (1 + np.exp(c * (x - d)))
+
+
+# Lower and upper interval boundaries for the three parameter values.
+bounds = [(0, None), (0, None), (None, None)]
+logistics_bounds = [(0, None), (0, None), (None, 0), (0, None)]
+
+power3 = DependenceFunction(_power3, bounds, latex="$a + b * x^c$")
+exp3 = DependenceFunction(_exp3, bounds, latex="$a + b * \exp(c * x)$")
+logistics4 = DependenceFunction(
+    _logistics4,
+    logistics_bounds,
+    weights=lambda x, y: y,
+    latex="$a + b / (1 + \exp[c * (x -d)])$",
+)
+alpha3 = DependenceFunction(
+    _alpha3,
+    bounds,
+    d_of_x=logistics4,
+    weights=lambda x, y: y,
+    latex="$(a + b * x^c) / 2.0445^{1 / F()}$",
+)
+
+# Define the structure of the joint distribution.
+
+# Wind speed.
+dist_description_0 = {
+    "distribution": ExponentiatedWeibullDistribution(),
+    "intervals": WidthOfIntervalSlicer(2, min_n_points=50),
+}
+# Wave height.
+dist_description_1 = {
+    "distribution": ExponentiatedWeibullDistribution(f_delta=5),
+    "intervals": WidthOfIntervalSlicer(0.5),
+    "conditional_on": 0,
+    "parameters": {"alpha": alpha3, "beta": logistics4,},
+}
+# Zero-up-crossing period.
+dist_description_2 = {
+    "distribution": LogNormalDistribution(),
+    "conditional_on": 1,
+    "parameters": {"mu": power3, "sigma": exp3},
+}
+
+
+model = GlobalHierarchicalModel(
+    [dist_description_0, dist_description_1, dist_description_2]
+)
+
+# Define a dictionary that describes the model.
+semantics = {
+    "names": ["Wind speed", "Significant wave height", "Zero-up-crossing period"],
+    "symbols": ["V", "H_s", "T_z"],
+    "units": ["m/s", "m", "s"],
+}
+
+# Fit the model to the data (estimate the model's parameter values).
+model.fit(data)
+
+# Print the estimated parameter values.
+print(model)
+
+# Create plots to inspect the model's goodness-of-fit.
+fig1, axs = plt.subplots(1, 3, figsize=[18, 7.2])
+plot_marginal_quantiles(model, data, semantics, axes=axs)
+fig2, axs = plt.subplots(1, 4, figsize=[22, 7.2])
+plot_dependence_functions(model, semantics, axes=axs)
+
+
+# Set up the three-dimensional mesh-grid for the 3D surface.
+v_step = 2.0
+h_step = 0.4
+t_step = 0.4
+vgrid, h, t = np.mgrid[0:50:v_step, 0:22:h_step, 0:22:t_step]
+f = np.empty_like(vgrid)
+
+for i in range(vgrid.shape[0]):
+    for j in range(vgrid.shape[1]):
+        for k in range(vgrid.shape[2]):
+            f[i, j, k] = model.pdf([vgrid[i, j, k], h[i, j, k], t[i, j, k]])
+print("Done with calculating pdf")
+
+
+# Calculate 3D Contour.
+state_duration = 1  # hours
+return_period = 20  # years
+alpha = state_duration / (return_period * 365.25 * 24)
+HDC = HighestDensityContour(model, alpha, limits=[(0, 50), (0, 25), (0, 25)])
+print("20-yr HDC has a density value of " + str(HDC.fm))
+iso_val = HDC.fm
+verts, faces, _, _ = marching_cubes(f, iso_val, spacing=(v_step, h_step, t_step))
+
+# Plot 3D Contour.
+fig = plt.figure()
```

Comment:
This kind of plot (trisurf) needs the vertices and faces of the contour. An easy alternative would be a scatter plot of the already calculated contour points. 
For example:

```python
#randomly select only 5% of the contour's points to increase performance
rng = np.random.default_rng(42)
n_contour_points = len(HDC.coordinates)
random_points = rng.choice(
    n_contour_points, int(0.05 * n_contour_points), replace=False
)

Xs = HDC.coordinates[random_points, 0]
Ys = HDC.coordinates[random_points, 1]
Zs = HDC.coordinates[random_points, 2]

fig = plt.figure()
ax = fig.add_subplot(111, projection="3d")
ax.scatter(Xs, Ys, Zs, c="#004488")
ax.set_xlabel("Wind speed (m/s)")
ax.set_ylabel("Significant wave height (m)")
ax.set_zlabel("Zero-up-crossing period (s)")
```

will create a plot like:
![grafik](https://user-images.githubusercontent.com/32780200/132243591-a263cb76-96c8-454b-b794-4f6246e04313.png)

Reply:
I think this is a great alternative. Because it allows us to avoid using scikit-image, I would prefer using a scatter plot versus the surface plot. Probably the script also runs faster as we do not need to compute the surface using marching_cubes.

@lenastroeer What do you think?

Reply:
Yes, that sounds perfect! I will include everything as Kai suggested!